services/QuillLMS/app/controllers/auth/google_controller.rb

```diff
@@ -1,28 +1,48 @@
 class Auth::GoogleController < ApplicationController
-  before_action :set_profile,                     only: :google
-  before_action :set_user,                        only: :google
-  before_action :save_teacher_from_google_signup, only: :google
-  before_action :save_student_from_google_signup, only: :google
-  before_action :follow_google_redirect,          only: :google
-
-  def google
-    if @user.teacher?
-      GoogleStudentImporterWorker.perform_async(@user.id, 'Auth::GoogleController')
-    end
+  before_action :set_profile, only: [:authorization_and_authentication, :authentication]
+  before_action :verify_credentials, only: :authentication
```

Comment:
When a user arrives on the `authorization_and_authentication` route, they have already been authorized (i.e. they have a valid refresh_token). When arriving via the `authentication` route, we don't actually know if their refresh_token exists or is valid, hence the callback only used for this route.

Maybe renaming it `verify_refresh_token` would be more clear?

Reply:
Actually, I'm going to call it `verify_authorization_for_offline_access` since that's the purpose of that check. We need authorization in order to perform the background jobs.

Reply:
Thanks for the explanation and I like the rename,